Datepicker with reactive form and strictTemplates throws Type 'string' is not assignable to type 'Date'

## Describe the bug

The datepicker component in its reactive form implementation as shown in the [official documentation](https://clarity.design/documentation/datepicker) at the *Reactive Form Demo* section, throws the following typescript compilation error, when `strictTemplates: true` in `tsconfig.json`:

```
error TS2322: Type 'string' is not assignable to type 'Date'.
```

The app is compiled successfully when `strictTemplates` is disabled. 

## How to reproduce

1. Ensure `strictTemplates` is enabled `tsconfig.json`
```tsconfig.json
{
...
  "angularCompilerOptions": {
    ...
    "strictTemplates": true
  }
}
```
2. Use the example code, exactly as found in the [official documentation](https://clarity.design/documentation/datepicker) at the *Reactive Form Demo* section.

3. See typescript compilation error
```
ERROR in ...component.html:5:26 - error TS2322: Type 'string' is not assignable to type 'Date'.
```

## Expected behavior

The app should compile successfully as it does when `strictTemplates` is not enabled.

## Versions

**App**

- Angular: [9.1.11]
- Node: [14.4.0]
- Clarity: [3.1.4]

**Device:**

- Type: [MacBook]
- OS: [iOS > docker > node:14-alpine]
- Browser [chrome]
- Version [latest]

--- a/packages/angular/golden/clr-angular.d.ts
+++ b/packages/angular/golden/clr-angular.d.ts
@@ -68,7 +68,7 @@ export declare abstract class ClrAbstractContainer implements DynamicWrapper, On
 }
 
 export declare class ClrAccordion implements OnInit, OnChanges, AfterViewInit, OnDestroy {
-    multiPanel: boolean;
+    multiPanel: boolean | string;
     panels: QueryList<ClrAccordionPanel>;
     subscriptions: Subscription[];
     constructor(accordionService: AccordionService);
@@ -306,8 +306,8 @@ export declare class ClrCombobox<T> extends WrappedFormControl<ClrComboboxContai
     get id(): string;
     protected index: number;
     invalid: boolean;
-    set multiSelect(value: boolean);
-    get multiSelect(): boolean;
+    set multiSelect(value: boolean | string);
+    get multiSelect(): boolean | string;
     get openState(): boolean;
     optionSelected: ClrOptionSelected<T>;
     optionSelectionService: OptionSelectionService<T>;
@@ -660,7 +660,7 @@ export declare class ClrDatagridHideableColumn implements OnDestroy {
     set clrDgHidden(hidden: boolean);
     set clrDgHideableColumn(value: {
         hidden: boolean;
-    });
+    } | string);
     hiddenChange: EventEmitter<boolean>;
     constructor(titleTemplateRef: TemplateRef<any>, viewContainerRef: ViewContainerRef, columnsService: ColumnsService, columnState: BehaviorSubject<ColumnState>);
     ngOnDestroy(): void;
@@ -871,7 +871,7 @@ export declare class ClrDateContainer implements DynamicWrapper, OnDestroy, Afte
 
 export declare class ClrDateInput extends WrappedFormControl<ClrDateContainer> implements OnInit, AfterViewInit, OnDestroy {
     protected control: NgControl;
-    set date(date: Date);
+    set date(date: Date | string);
     dateChange: EventEmitter<Date>;
     set disabled(value: boolean | string);
     get disabled(): boolean | string;
@@ -889,7 +889,7 @@ export declare class ClrDateInput extends WrappedFormControl<ClrDateContainer> i
     onValueChange(target: HTMLInputElement): void;
     setFocusStates(): void;
     triggerValidation(): void;
-    static ngAcceptInputType_date: Date | null;
+    static ngAcceptInputType_date: Date | null | string;
 }
 
 export declare class ClrDatepickerModule {
@@ -1049,7 +1049,7 @@ export declare class ClrExpandableAnimation {
 }
 
 export declare class ClrFocusOnViewInit implements AfterViewInit {
-    set isEnabled(value: boolean);
+    set isEnabled(value: boolean | string);
     constructor(el: ElementRef, platformId: any, focusOnViewInit: boolean, document: any, renderer: Renderer2);
     ngAfterViewInit(): void;
     onFocusout(): void;
@@ -1097,8 +1097,8 @@ export declare class ClrIconModule {
 }
 
 export declare class ClrIfActive implements OnDestroy {
-    set active(value: boolean);
-    get active(): boolean;
+    set active(value: boolean | string);
+    get active(): boolean | string;
     activeChange: EventEmitter<boolean>;
     constructor(ifActiveService: IfActiveService, id: number, template: TemplateRef<any>, container: ViewContainerRef);
     ngOnDestroy(): void;
@@ -1125,8 +1125,8 @@ export declare class ClrIfError extends AbstractIfState {
 }
 
 export declare class ClrIfExpanded implements OnInit, OnDestroy {
-    get expanded(): boolean;
-    set expanded(value: boolean);
+    get expanded(): boolean | string;
+    set expanded(value: boolean | string);
     expandedChange: EventEmitter<boolean>;
     constructor(template: TemplateRef<any>, container: ViewContainerRef, el: ElementRef, renderer: Renderer2, expand: IfExpandService);
     ngOnDestroy(): void;
@@ -1170,7 +1170,7 @@ export declare class ClrLabel implements OnInit, OnDestroy {
 }
 
 export declare class ClrLayout implements OnInit {
-    layout: ClrFormLayout;
+    layout: ClrFormLayout | string;
     layoutService: LayoutService;
     constructor(layoutService: LayoutService);
     ngOnInit(): void;
@@ -1180,11 +1180,11 @@ export declare class ClrLayoutModule {
 }
 
 export declare class ClrLoading implements OnDestroy {
-    get loadingState(): boolean | ClrLoadingState;
-    set loadingState(value: boolean | ClrLoadingState);
+    get loadingState(): boolean | string | ClrLoadingState;
+    set loadingState(value: boolean | string | ClrLoadingState);
     constructor(listener: LoadingListener);
     ngOnDestroy(): void;
-    static ngAcceptInputType_loadingState: boolean | ClrLoadingState | null;
+    static ngAcceptInputType_loadingState: boolean | ClrLoadingState | null | string;
 }
 
 export declare class ClrLoadingButton implements LoadingListener {
@@ -1444,11 +1444,11 @@ export declare class ClrProgressBar {
     get id(): string;
     get labeledClass(): boolean;
     get loopClass(): boolean;
-    max: number;
+    max: number | string;
     get progressClass(): boolean;
     get successClass(): boolean;
-    get value(): number;
-    set value(value: number);
+    get value(): number | string;
+    set value(value: number | string);
     displayStringValue(): boolean;
 }
 
@@ -1736,8 +1736,8 @@ export declare class ClrTabs implements AfterContentInit, OnDestroy {
     get isCurrentInOverflow(): boolean;
     get isVertical(): boolean;
     keyFocus: ClrKeyFocus;
-    set layout(layout: TabsLayout);
-    get layout(): TabsLayout;
+    set layout(layout: TabsLayout | string);
+    get layout(): TabsLayout | string;
     get tabIds(): string;
     get tabLinkDirectives(): ClrTabLink[];
     tabLinkElements: HTMLElement[];
@@ -2058,7 +2058,7 @@ export declare class ClrWizardPage implements OnInit {
     pageNavTitle: ClrWizardPageNavTitle;
     pageOnCancel: EventEmitter<ClrWizardPage>;
     pageTitle: ClrWizardPageTitle;
-    preventDefault: boolean;
+    preventDefault: boolean | string;
     previousButtonClicked: EventEmitter<ClrWizardPage>;
     get previousCompleted(): boolean;
     get previousStepDisabled(): boolean;
@@ -2214,7 +2214,7 @@ export declare const IS_TOGGLE_PROVIDER: {
 export declare function isToggleFactory(): BehaviorSubject<boolean>;
 
 export declare abstract class LoadingListener {
-    abstract loadingStateChange(state: ClrLoadingState): void;
+    abstract loadingStateChange(state: ClrLoadingState | string): void;
 }
 
 export declare class MainContainerWillyWonka extends WillyWonka {

--- a/packages/angular/projects/clr-angular/src/accordion/accordion.ts
+++ b/packages/angular/projects/clr-angular/src/accordion/accordion.ts
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2016-2020 VMware, Inc. All Rights Reserved.
+ * Copyright (c) 2016-2021 VMware, Inc. All Rights Reserved.
  * This software is released under MIT license.
  * The full license information can be found in LICENSE in the root directory of this project.
  */
@@ -31,7 +31,7 @@ import { AccordionStrategy } from './enums/accordion-strategy.enum';
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
 export class ClrAccordion implements OnInit, OnChanges, AfterViewInit, OnDestroy {
-  @Input('clrAccordionMultiPanel') multiPanel = false;
+  @Input('clrAccordionMultiPanel') multiPanel: boolean | string = false;
   @ContentChildren(ClrAccordionPanel, { descendants: true })
   panels: QueryList<ClrAccordionPanel>;
   subscriptions: Subscription[] = [];

--- a/packages/angular/projects/clr-angular/src/data/datagrid/datagrid-hideable-column.ts
+++ b/packages/angular/projects/clr-angular/src/data/datagrid/datagrid-hideable-column.ts
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2016-2019 VMware, Inc. All Rights Reserved.
+ * Copyright (c) 2016-2021 VMware, Inc. All Rights Reserved.
  * This software is released under MIT license.
  * The full license information can be found in LICENSE in the root directory of this project.
  */
@@ -54,7 +54,6 @@ export class ClrDatagridHideableColumn implements OnDestroy {
    * It allows the user to pre-configure the column's hide/show state. { hidden: true }
    * It's more verbose but has more Clarity.
    *
-   *
    * @example
    * *clrDgHideableColumn
    * *clrDgHideableColumn={hidden: false}
@@ -62,7 +61,11 @@ export class ClrDatagridHideableColumn implements OnDestroy {
    *
    */
   @Input('clrDgHideableColumn')
-  set clrDgHideableColumn(value: { hidden: boolean }) {
+  set clrDgHideableColumn(value: { hidden: boolean } | string) {
+    if (typeof value === 'string') {
+      this.clrDgHidden = false;
+      return;
+    }
     this.clrDgHidden = value && value.hidden ? value.hidden : false;
   }
 

--- a/packages/angular/projects/clr-angular/src/forms/combobox/combobox.ts
+++ b/packages/angular/projects/clr-angular/src/forms/combobox/combobox.ts
@@ -1,5 +1,5 @@
 /**
- * Copyright (c) 2016-2020 VMware, Inc. All Rights Reserved.
+ * Copyright (c) 2016-2021 VMware, Inc. All Rights Reserved.
  * This software is released under MIT license.
  * The full license information can be found in LICENSE in the root directory of this project.
  */
@@ -178,7 +178,7 @@ export class ClrCombobox<T> extends WrappedFormControl<ClrComboboxContainer>
   @Input('placeholder') placeholder = '';
 
   @Input('clrMulti')
-  set multiSelect(value: boolean) {
+  set multiSelect(value: boolean | string) {
     if (value) {
       this.optionSelectionService.selectionModel = new MultiSelectComboboxModel<T>();
     } else {

--- a/packages/angular/projects/clr-angular/src/forms/common/layout.ts
+++ b/packages/angular/projects/clr-angular/src/forms/common/layout.ts
@@ -1,5 +1,5 @@
 /**
- * Copyright (c) 2016-2020 VMware, Inc. All Rights Reserved.
+ * Copyright (c) 2016-2021 VMware, Inc. All Rights Reserved.
  * This software is released under MIT license.
  * The full license information can be found in LICENSE in the root directory of this project.
  */
@@ -11,7 +11,7 @@ import { ClrFormLayout, LayoutService } from './providers/layout.service';
   selector: '[clrForm][clrLayout]',
 })
 export class ClrLayout implements OnInit {
-  @Input('clrLayout') layout: ClrFormLayout;
+  @Input('clrLayout') layout: ClrFormLayout | string;
 
   constructor(public layoutService: LayoutService) {}
 

--- a/packages/angular/projects/clr-angular/src/forms/common/providers/layout.service.ts
+++ b/packages/angular/projects/clr-angular/src/forms/common/providers/layout.service.ts
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2016-2020 VMware, Inc. All Rights Reserved.
+ * Copyright (c) 2016-2021 VMware, Inc. All Rights Reserved.
  * This software is released under MIT license.
  * The full license information can be found in LICENSE in the root directory of this project.
  */
@@ -16,7 +16,7 @@ export enum ClrFormLayout {
 export class LayoutService {
   readonly minLabelSize = 1;
   readonly maxLabelSize = 12;
-  layout: ClrFormLayout = ClrFormLayout.HORIZONTAL;
+  layout: ClrFormLayout | string = ClrFormLayout.HORIZONTAL;
 
   // This is basically a replacement for Object.values(), which IE11 and Node <9 don't support :(
   // String enums cannot be reverse-mapped, meaning ClrFormLayout['COMPACT'] does not return 'compact' so

--- a/packages/angular/projects/clr-angular/src/forms/datepicker/date-input.ts
+++ b/packages/angular/projects/clr-angular/src/forms/datepicker/date-input.ts
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2016-2020 VMware, Inc. All Rights Reserved.
+ * Copyright (c) 2016-2021 VMware, Inc. All Rights Reserved.
  * This software is released under MIT license.
  * The full license information can be found in LICENSE in the root directory of this project.
  */
@@ -54,18 +54,18 @@ import { isBooleanAttributeSet } from '../../utils/component/is-boolean-attribut
   providers: [DatepickerFocusService],
 })
 export class ClrDateInput extends WrappedFormControl<ClrDateContainer> implements OnInit, AfterViewInit, OnDestroy {
-  public static ngAcceptInputType_date: Date | null;
+  public static ngAcceptInputType_date: Date | null | string;
 
   @Input() placeholder: string;
   @Output('clrDateChange') dateChange: EventEmitter<Date> = new EventEmitter<Date>(false);
   @Input('clrDate')
-  set date(date: Date) {
+  set date(date: Date | string) {
     if (this.previousDateChange !== date) {
-      this.updateDate(this.getValidDateValueFromDate(date));
+      this.updateDate(this.getValidDateValueFromDate(date as Date));
     }
 
     if (!this.initialClrDateInputValue) {
-      this.initialClrDateInputValue = date;
+      this.initialClrDateInputValue = date as Date;
     }
   }
 

--- a/packages/angular/projects/clr-angular/src/layout/tabs/providers/tabs.service.ts
+++ b/packages/angular/projects/clr-angular/src/layout/tabs/providers/tabs.service.ts
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2016-2019 VMware, Inc. All Rights Reserved.
+ * Copyright (c) 2016-2021 VMware, Inc. All Rights Reserved.
  * This software is released under MIT license.
  * The full license information can be found in LICENSE in the root directory of this project.
  */
@@ -11,7 +11,7 @@ import { TabsLayout } from '../enums/tabs-layout.enum';
 export class TabsService {
   private _children: ClrTab[] = [];
 
-  layout: TabsLayout = TabsLayout.HORIZONTAL;
+  layout: TabsLayout | string = TabsLayout.HORIZONTAL;
 
   register(tab: ClrTab) {
     this._children.push(tab);

--- a/packages/angular/projects/clr-angular/src/layout/tabs/tabs.ts
+++ b/packages/angular/projects/clr-angular/src/layout/tabs/tabs.ts
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2016-2020 VMware, Inc. All Rights Reserved.
+ * Copyright (c) 2016-2021 VMware, Inc. All Rights Reserved.
  * This software is released under MIT license.
  * The full license information can be found in LICENSE in the root directory of this project.
  */
@@ -108,7 +108,7 @@ export class ClrTabs implements AfterContentInit, OnDestroy {
   }
 
   @Input('clrLayout')
-  set layout(layout: TabsLayout) {
+  set layout(layout: TabsLayout | string) {
     if (
       Object.keys(TabsLayout)
         .map(key => {
@@ -119,7 +119,7 @@ export class ClrTabs implements AfterContentInit, OnDestroy {
       this.tabsService.layout = layout;
     }
   }
-  get layout(): TabsLayout {
+  get layout(): TabsLayout | string {
     return this.tabsService.layout;
   }
 

--- a/packages/angular/projects/clr-angular/src/progress/progress-bars/progress-bar.ts
+++ b/packages/angular/projects/clr-angular/src/progress/progress-bars/progress-bar.ts
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2016-2020 VMware, Inc. All Rights Reserved.
+ * Copyright (c) 2016-2021 VMware, Inc. All Rights Reserved.
  * This software is released under MIT license.
  * The full license information can be found in LICENSE in the root directory of this project.
  */
@@ -29,16 +29,23 @@ export class ClrProgressBar {
   }
 
   // Progress
-  @Input('clrMax') max = 100;
+  @Input('clrMax') max: number | string = 100;
   @Input('clrDisplayval') displayval: string;
 
-  private _value = 0;
+  private _value: number | string = 0;
   @Input('clrValue')
   get value() {
     return this._value;
   }
 
-  set value(value) {
+  set value(value: number | string) {
+    /**
+     * No need to convert to `number` cause we could have
+     * floating point and parseInt will round the numbers
+     *
+     * working with string won't have any side-effects,
+     * we don't do any math so string will do the job.
+     */
     this._value = value;
   }
 

--- a/packages/angular/projects/clr-angular/src/utils/conditional/if-active.directive.ts
+++ b/packages/angular/projects/clr-angular/src/utils/conditional/if-active.directive.ts
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2016-2020 VMware, Inc. All Rights Reserved.
+ * Copyright (c) 2016-2021 VMware, Inc. All Rights Reserved.
  * This software is released under MIT license.
  * The full license information can be found in LICENSE in the root directory of this project.
  */
@@ -64,7 +64,7 @@ export class ClrIfActive implements OnDestroy {
    * @param value
    */
   @Input('clrIfActive')
-  public set active(value: boolean) {
+  public set active(value: boolean | string) {
     if (value) {
       this.ifActiveService.current = this.id;
     }

--- a/packages/angular/projects/clr-angular/src/utils/conditional/if-expanded.directive.ts
+++ b/packages/angular/projects/clr-angular/src/utils/conditional/if-expanded.directive.ts
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2016-2020 VMware, Inc. All Rights Reserved.
+ * Copyright (c) 2016-2021 VMware, Inc. All Rights Reserved.
  * This software is released under MIT license.
  * The full license information can be found in LICENSE in the root directory of this project.
  */
@@ -24,12 +24,12 @@ import { IfExpandService } from './if-expanded.service';
 export class ClrIfExpanded implements OnInit, OnDestroy {
   private _expanded = false;
 
-  get expanded(): boolean {
+  get expanded(): boolean | string {
     return this._expanded;
   }
 
   @Input('clrIfExpanded')
-  set expanded(value: boolean) {
+  set expanded(value: boolean | string) {
     if (typeof value === 'boolean') {
       this.expand.expanded = value;
       this._expanded = value;

--- a/packages/angular/projects/clr-angular/src/utils/focus-trap/focus-trap.directive.ts
+++ b/packages/angular/projects/clr-angular/src/utils/focus-trap/focus-trap.directive.ts
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2016-2020 VMware, Inc. All Rights Reserved.
+ * Copyright (c) 2016-2021 VMware, Inc. All Rights Reserved.
  * This software is released under MIT license.
  * The full license information can be found in LICENSE in the root directory of this project.
  */
@@ -51,8 +51,8 @@ export class FocusTrapDirective implements OnInit, AfterViewInit, OnDestroy {
     strict: true,
   };
   @Input('clrFocusTrap')
-  set config(config: FocusTrapConfig) {
-    this._config = Object.assign(this._config, config);
+  set config(config: FocusTrapConfig | string) {
+    this._config = Object.assign(this._config, config === '' ? {} : config);
   }
 
   @HostListener('document:focusin', ['$event'])

--- a/packages/angular/projects/clr-angular/src/utils/focus/focus-on-view-init/focus-on-view-init.ts
+++ b/packages/angular/projects/clr-angular/src/utils/focus/focus-on-view-init/focus-on-view-init.ts
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2016-2020 VMware, Inc. All Rights Reserved.
+ * Copyright (c) 2016-2021 VMware, Inc. All Rights Reserved.
  * This software is released under MIT license.
  * The full license information can be found in LICENSE in the root directory of this project.
  */
@@ -53,7 +53,7 @@ export class ClrFocusOnViewInit implements AfterViewInit {
 
   private _isEnabled: boolean;
   @Input('clrFocusOnViewInit')
-  set isEnabled(value: boolean) {
+  set isEnabled(value: boolean | string) {
     if (this.focusOnViewInit && typeof value === 'boolean') {
       this._isEnabled = value;
     }

--- a/packages/angular/projects/clr-angular/src/utils/focus/key-focus/key-focus.ts
+++ b/packages/angular/projects/clr-angular/src/utils/focus/key-focus/key-focus.ts
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2016-2020 VMware, Inc. All Rights Reserved.
+ * Copyright (c) 2016-2021 VMware, Inc. All Rights Reserved.
  * This software is released under MIT license.
  * The full license information can be found in LICENSE in the root directory of this project.
  */
@@ -28,7 +28,7 @@ import { preventArrowKeyScroll, keyValidator } from './util';
 })
 export class ClrKeyFocus {
   constructor(private elementRef: ElementRef) {}
-  @Input('clrDirection') direction = ClrFocusDirection.VERTICAL;
+  @Input('clrDirection') direction: ClrFocusDirection | string = ClrFocusDirection.VERTICAL;
   @Input('clrFocusOnLoad') focusOnLoad = false;
   @Output('clrFocusChange') private focusChange: EventEmitter<number> = new EventEmitter<number>();
   @ContentChildren(ClrKeyFocusItem, { descendants: true })
@@ -36,12 +36,16 @@ export class ClrKeyFocus {
 
   private _focusableItems: Array<FocusableItem>;
   @Input('clrKeyFocus')
-  set focusableItems(elements: Array<FocusableItem>) {
+  /**
+   * Here we use `any` cause any other type require reworking all methods below and a lot of more ifs.
+   * this method will only work with array with FocusableItems anyway so any other value will be ignored.
+   */
+  set focusableItems(elements: Array<FocusableItem> | any) {
     // We accept a list of focusable elements (HTMLElements or existing Directives) or auto query for clrKeyFocusItem
     // We accept a list reference in the cases where we cannot use ContentChildren to query
     // ContentChildren can be unavailable if content is projected outside the scope of the component (see tabs).
-    if (elements && elements.length) {
-      this._focusableItems = elements;
+    if (Array.isArray(elements) && elements.length) {
+      this._focusableItems = elements as Array<FocusableItem>;
       this.initializeFocus();
     }
   }

--- a/packages/angular/projects/clr-angular/src/utils/loading/loading-listener.ts
+++ b/packages/angular/projects/clr-angular/src/utils/loading/loading-listener.ts
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2016-2018 VMware, Inc. All Rights Reserved.
+ * Copyright (c) 2016-2021 VMware, Inc. All Rights Reserved.
  * This software is released under MIT license.
  * The full license information can be found in LICENSE in the root directory of this project.
  */
@@ -11,5 +11,5 @@ import { ClrLoadingState } from './loading';
  * This does not mean you should extend it, simply implementing it is fine.
  */
 export abstract class LoadingListener {
-  abstract loadingStateChange(state: ClrLoadingState): void;
+  abstract loadingStateChange(state: ClrLoadingState | string): void;
 }

--- a/packages/angular/projects/clr-angular/src/utils/loading/loading.ts
+++ b/packages/angular/projects/clr-angular/src/utils/loading/loading.ts
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2016-2018 VMware, Inc. All Rights Reserved.
+ * Copyright (c) 2016-2021 VMware, Inc. All Rights Reserved.
  * This software is released under MIT license.
  * The full license information can be found in LICENSE in the root directory of this project.
  */
@@ -16,19 +16,19 @@ export enum ClrLoadingState {
 
 @Directive({ selector: '[clrLoading]' })
 export class ClrLoading implements OnDestroy {
-  public static ngAcceptInputType_loadingState: boolean | ClrLoadingState | null;
+  public static ngAcceptInputType_loadingState: boolean | ClrLoadingState | null | string;
 
   // We find the first parent that handles something loading
   constructor(@Optional() private listener: LoadingListener) {}
 
-  private _loadingState: ClrLoadingState = ClrLoadingState.DEFAULT;
+  private _loadingState: ClrLoadingState | string = ClrLoadingState.DEFAULT;
 
   public get loadingState() {
     return this._loadingState;
   }
 
   @Input('clrLoading')
-  public set loadingState(value: boolean | ClrLoadingState) {
+  public set loadingState(value: boolean | string | ClrLoadingState) {
     if (value === true) {
       value = ClrLoadingState.LOADING;
     } else if (!value) {

--- a/packages/angular/projects/clr-angular/src/wizard/wizard-page.ts
+++ b/packages/angular/projects/clr-angular/src/wizard/wizard-page.ts
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2016-2020 VMware, Inc. All Rights Reserved.
+ * Copyright (c) 2016-2021 VMware, Inc. All Rights Reserved.
  * This software is released under MIT license.
  * The full license information can be found in LICENSE in the root directory of this project.
  */
@@ -235,7 +235,7 @@ export class ClrWizardPage implements OnInit {
    * @memberof WizardPage
    *
    */
-  @Input('clrWizardPagePreventDefault') public preventDefault = false;
+  @Input('clrWizardPagePreventDefault') public preventDefault: boolean | string = false;
 
   /**
    *

--- a/packages/angular/projects/dev/src/app/datagrid/pagination/pagination.ts
+++ b/packages/angular/projects/dev/src/app/datagrid/pagination/pagination.ts
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2016-2020 VMware, Inc. All Rights Reserved.
+ * Copyright (c) 2016-2021 VMware, Inc. All Rights Reserved.
  * This software is released under MIT license.
  * The full license information can be found in LICENSE in the root directory of this project.
  */
@@ -25,8 +25,8 @@ export class DatagridPaginationDemo {
     this.users = inventory.all;
   }
 
-  pageChange($event: Event) {
-    console.log($event);
+  pageChange(pageNumber: number) {
+    console.log(pageNumber);
   }
 
   toggleCurrentPageInput() {

--- a/packages/angular/projects/dev/src/app/typography/typography-font-autopsy.html
+++ b/packages/angular/projects/dev/src/app/typography/typography-font-autopsy.html
@@ -1,5 +1,5 @@
 <!--
-    ~ Copyright (c) 2016-2020 VMware, Inc. All Rights Reserved.
+    ~ Copyright (c) 2016-2021 VMware, Inc. All Rights Reserved.
     ~ This software is released under MIT license.
     ~ The full license information can be found in LICENSE in the root directory of this project.
 -->
@@ -30,7 +30,7 @@
       <form clrForm (submit)="noop()" [formGroup]="model" clrLayout="vertical">
         <clr-input-container>
           <label>Font</label>
-          <input clrInput name="font" formControlName="font" (change)="checkForPreset($event.target.value)" />
+          <input clrInput name="font" formControlName="font" (change)="checkForPreset($event)" />
         </clr-input-container>
 
         <clr-input-container>

--- a/packages/angular/projects/dev/src/app/typography/typography-font-autopsy.ts
+++ b/packages/angular/projects/dev/src/app/typography/typography-font-autopsy.ts
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2016-2020 VMware, Inc. All Rights Reserved.
+ * Copyright (c) 2016-2021 VMware, Inc. All Rights Reserved.
  * This software is released under MIT license.
  * The full license information can be found in LICENSE in the root directory of this project.
  */
@@ -36,7 +36,8 @@ export class TypographyFontAutopsyDemo {
 
   fontField = 'Metropolis';
 
-  checkForPreset(fontName: string) {
+  checkForPreset(event: Event) {
+    const fontName = (event.target as HTMLInputElement).value;
     const sanitizedFontName = fontName.replace(/["']/g, '');
     const preset = getPreset(sanitizedFontName);
 

--- a/packages/angular/projects/dev/src/app/typography/utils/font-switcher.html
+++ b/packages/angular/projects/dev/src/app/typography/utils/font-switcher.html
@@ -1,5 +1,5 @@
 <!--
-    ~ Copyright (c) 2016-2020 VMware, Inc. All Rights Reserved.
+    ~ Copyright (c) 2016-2021 VMware, Inc. All Rights Reserved.
     ~ This software is released under MIT license.
     ~ The full license information can be found in LICENSE in the root directory of this project.
 -->
@@ -7,7 +7,7 @@
 <form clrForm (submit)="noop()" [formGroup]="presetSwitcher" [clrLayout]="layout" style="padding: 0;">
   <clr-select-container>
     <label>{{label}}</label>
-    <select clrSelect formControlName="preset" name="formSwitcherSelect" (change)="switchFont($event.target.value)">
+    <select clrSelect formControlName="preset" name="formSwitcherSelect" (change)="handleSwitchFont($event)">
       <option [value]="">-- None --</option>
       <option *ngFor="let preset of presets" [value]="preset">{{preset}}</option>
     </select>

--- a/packages/angular/projects/dev/src/app/typography/utils/font-switcher.ts
+++ b/packages/angular/projects/dev/src/app/typography/utils/font-switcher.ts
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2016-2020 VMware, Inc. All Rights Reserved.
+ * Copyright (c) 2016-2021 VMware, Inc. All Rights Reserved.
  * This software is released under MIT license.
  * The full license information can be found in LICENSE in the root directory of this project.
  */
@@ -39,7 +39,11 @@ export class FontSwitcher {
     this.switchFont(val);
   }
 
-  switchFont(presetToLoad: string) {
+  public handleSwitchFont(event: Event) {
+    this.switchFont((event.target as HTMLInputElement).value);
+  }
+
+  private switchFont(presetToLoad: string) {
     if (checkForExistingPresetName(presetToLoad)) {
       this.presetSwitcher.setValue({ preset: presetToLoad });
       this.sendFontSwitch(presetToLoad);

--- a/packages/angular/tsconfig.json
+++ b/packages/angular/tsconfig.json
@@ -29,6 +29,7 @@
   },
   "angularCompilerOptions": {
     "fullTemplateTypeCheck": true,
-    "strictInjectionParameters": true
+    "strictInjectionParameters": true,
+    "strictTemplates": true
   }
 }

